Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -28,7 +28,12 @@ def edit
       EOS
     end
 
-    paths = args.named.to_formulae_paths.presence
+    paths = args.named.to_formulae_paths.select do |path|
+      formula_name = File.basename(path.basename, ".*")
+      Formulary.factory(formula_name)
+      rescue FormulaUnavailableError => e
```

Comment:
This will trigger on syntax errors. Use `path.exist?` instead.

Reply:
@MikeMcQuaid 
`path.exist?` returns the boolean value and cannot cause `FormulaUnavailableError` and show `No available formula with the name [Formula] ` like below, 
therefore I would like to cause `FormulaUnavailableError` by `Formulary#factory` or `NamedArgs#to_formulae`

```bash
$brew edit abcprogram  
Usage: brew edit [formula]

Open formula in the editor set by EDITOR or HOMEBREW_EDITOR, or open the
Homebrew repository for editing if no formula is provided.

  -d, --debug                      Display any debugging information.
  -q, --quiet                      Suppress any warnings.
  -v, --verbose                    Make some output more verbose.
  -h, --help                       Show this message.

Error: Invalid usage: No available formula with the name "abcprogram" . Run brew create $URL to create a new Formula!
```

Reply:
You can craft that message by hand if desired (although I think it's nicer to state that it doesn't exist on disk than stating there's "no available formula"). Exceptions should be exceptional: it's not desirable to intentionally try to trigger an exception only to catch it just for the message.